src/test/java/uk/gov/justice/hmpps/prison/util/builders/dsl/NomisData.kt

```diff
@@ -1,16 +1,27 @@
 package uk.gov.justice.hmpps.prison.util.builders.dsl
 
-import uk.gov.justice.hmpps.prison.util.builders.TestDataContext
+import org.springframework.security.authentication.UsernamePasswordAuthenticationToken
+import org.springframework.security.core.context.SecurityContextHolder
+import org.springframework.security.core.userdetails.User
+import org.springframework.stereotype.Component
 import uk.gov.justice.hmpps.prison.util.builders.randomName
 import java.time.LocalDate
 
-class NomisDataBuilder(
-  testDataContext: TestDataContext,
-) {
-  private val offenderBuilderFactory: OffenderBuilderFactory = OffenderBuilderFactory(testDataContext = testDataContext)
-  fun build(dsl: NomisData.() -> Unit) = NomisData(
-    offenderBuilderFactory,
-  ).apply(dsl)
+@Component
+class NomisDataBuilder(private val offenderBuilderFactory: OffenderBuilderFactory) {
+  fun build(dsl: NomisData.() -> Unit) {
+    SecurityContextHolder.setContext(
```

Comment:
this would effectively be done via the webClient call, but since we are bypassing this set the context directly with a hardcoded user (with no roles). For now, this seems "good enough" so long as the test data doesn't require a specific user - or the service makes any further authentication calls.

Reply:
Nice. I think there will be some user based tests coming for which we could just make user oart if the top level DSL.